### ISSUE TITLE
Implement a few basic tests

### ISSUE THREAD

**Issue by mfherbst** [andrenarchy/libfulltext#30]
This isssue is intended to be a discussion thread for a first basic test system.

**Comment by mfherbst** [comment]
One idea I had floating around would be to download a fulltext pdf and compare it with a stored hash, say sha1 for consistency.

**Comment by andrenarchy** [comment]
First unit test with SHA1 comparison implemented in branch https://github.com/andrenarchy/libfulltext/tree/feature/30-add-unit-tests, see https://travis-ci.org/andrenarchy/libfulltext/jobs/323660158. Will add more tomorrow – now :tada:!

**Comment by mfherbst** [comment]
Looks good to me. Not sure I would have the `test_utils.py` in the same directory as the main project. Maybe we should have a `tests` folder or `tests` subfolder? 

Btw: I really like how taking the handler as a function pays off in this case ;).

**Comment by andrenarchy** [comment]
Yeah, we could move the `test_utils.py` somewhere else. I just think we should keep the test files for modules (like `test_springer.py` for `springer.py`) right next to the module file. Otherwise things tend to get out of sync quickly.

**Comment by mfherbst** [comment]
Agree. That sounds reasonable.

**Comment by mfherbst** [comment]
In my opinion our first test setup would need (feel free to edit):
- [X] Unit tests for retrieving pdf fulltexts
- [ ] Unit test for resolving doi to RA and to publisher
- [ ] Functionality test running `get_fulltext.py` directly with commandline args.
